src/Service/HealthCheck/ComprehensiveHealthReportResponseWriter.cs

```diff
@@ -74,20 +80,66 @@ public Task WriteResponse(HttpContext context)
-                LogTrace($"Health check response writer writing status as: {dabHealthCheckReport.Status}");
-                return context.Response.WriteAsync(response);
+                string? response;
+                // Check if the cache is enabled 
+                if (config.CacheTtlSecondsForHealthReport != null && config.CacheTtlSecondsForHealthReport > 0)
+                {
+                    if (!_cache.TryGetValue(CACHE_KEY, out response))
+                    {
+                        ComprehensiveHealthCheckReport dabHealthCheckReport = _healthCheckHelper.GetHealthCheckResponse(context, config);
+
+                        response = JsonSerializer.Serialize(dabHealthCheckReport, options: new JsonSerializerOptions { WriteIndented = true, DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull });
+                        LogTrace($"Health check response writer writing status as: {dabHealthCheckReport.Status}");
+
+                        try
+                        {
+                            // Cache the response for cache-ttl-seconds provided by the user in the config
+                            MemoryCacheEntryOptions cacheEntryOptions = new MemoryCacheEntryOptions()
+                                .SetAbsoluteExpiration(TimeSpan.FromSeconds((double)config.CacheTtlSecondsForHealthReport));
+
+                            _cache.Set(CACHE_KEY, response, cacheEntryOptions);
+                            LogTrace($"Health check response writer writing status as: {dabHealthCheckReport.Status}");
+                        }
+                        catch (Exception ex)
+                        {
+                            LogTrace($"Error in caching health check response: {ex.Message}");
+                        }
+                    }
+
+                    // Ensure cachedResponse is not null before calling WriteAsync
+                    if (response != null)
+                    {
+                        // Return the cached or newly generated response
+                        await context.Response.WriteAsync(response);
+                    }
+                    else
+                    {
+                        // Handle the case where cachedResponse is still null
+                        LogTrace("Error: The cached health check response is null.");
+                        context.Response.StatusCode = 500; // Internal Server Error
+                        await context.Response.WriteAsync("Failed to generate health check response.");
+                    }
+                }
+                else
+                {
+                    ComprehensiveHealthCheckReport dabHealthCheckReport = _healthCheckHelper.GetHealthCheckResponse(context, config);
```

Comment:
The code in this else block is repeated. Can be put in a common function to avoid duplication.